projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -22,41 +25,81 @@ import styles from './styles.module.scss';
  * @param {boolean} props.isPublicizeEnabled            - Whether Publicize is enabled for this post.
  * @param {boolean} props.isRePublicizeFeatureEnabled   - True if the RePublicize feature is available.
  * @param {boolean} props.isPublicizeDisabledBySitePlan - A combination of the republicize feature being enabled and/or the post not being published.
+ * @param {number} props.numberOfSharesRemaining        - The number of shares remaining for the current period. Optional.
  * @returns {object}                                    - Publicize form component.
  */
 export default function PublicizeForm( {
 	isPublicizeEnabled,
 	isRePublicizeFeatureEnabled,
 	isPublicizeDisabledBySitePlan,
+	numberOfSharesRemaining = null,
 } ) {
-	const { connections, toggleById, hasConnections } = useSocialMediaConnections();
+	const {
+		connections,
+		toggleById,
+		hasConnections,
+		enabledConnections,
+	} = useSocialMediaConnections();
 	const { message, updateMessage, maxLength } = useSocialMediaMessage();
 
 	const isDisabled = () =>
 		! isRePublicizeFeatureEnabled && connections.every( connection => ! connection.toggleable );
 	const Wrapper = isPublicizeDisabledBySitePlan ? Disabled : Fragment;
 
+	const outOfConnections =
+		numberOfSharesRemaining !== null && numberOfSharesRemaining <= enabledConnections.length;
+
 	return (
 		<Wrapper>
 			{ hasConnections && (
-				<PanelRow>
-					<ul className={ styles[ 'connections-list' ] }>
-						{ connections.map(
-							( { display_name, enabled, id, service_name, toggleable, profile_picture } ) => (
-								<PublicizeConnection
-									disabled={ isRePublicizeFeatureEnabled ? ! isPublicizeEnabled : ! toggleable }
-									enabled={ enabled && ! isPublicizeDisabledBySitePlan }
-									key={ id }
-									id={ id }
-									label={ display_name }
-									name={ service_name }
-									toggleConnection={ toggleById }
-									profilePicture={ profile_picture }
-								/>
-							)
-						) }
-					</ul>
-				</PanelRow>
+				<>
+					{ ! isDisabled() && numberOfSharesRemaining !== null && (
+						<PanelRow>
+							<Notice type={ numberOfSharesRemaining < connections.length ? 'warning' : 'default' }>
+								{ createInterpolateElement(
+									sprintf(
+										/* translators: %d is the number of shares remaining, upgradeLink is the link to upgrade to a different plan */
+										_n(
+											'You have %d share remaining. <upgradeLink>Upgrade</upgradeLink> to share to all your social media accounts.',
```

Comment:
Maybe for a follow up change but when I have more shares remaining than connections, the phrase 'to share to all your social media accounts' doesn't make as much sense because the post will be shared to all accounts. It should be 'to share this post as many times as you like' or 'to remove this limit'.

It's probably on the edge at the moment, but with this logic, should we make this a separate `ShareLimitNotice` component?

Reply:
Yeah, we can create a new component in a follow-up PR. Good idea 👍🏼